Car_Code/cheme_car_datalogging_codebase/cheme_car_datalogging_codebase.ino

```diff
@@ -4,12 +4,72 @@
 // IMU - Needs to be initialized, periodically poll sensor, store Z-axis angle in variable, take initial value as 0 deg print out to serial monitor, look into kalman filters for reducing sensor noise.
 // Stir Bar Motor - Needs to be initialized, just constantly turn at 80% speed for now, we'll tune speed later.
 // Servo Motor - Needs to be initialized, turn 180 deg clockwise once at start to dump reactants, wait 1 sec to finish dumping, turn back 180 deg counter-clockwise to return to upright position, set speed to 100% for now, tune later.
-// MicroSD Card - Store all variable data to MicroSD Card on microcontroller, allow for each run to be saved under a different filename (potentially in CSV format) so we can keep track of everything.
+// Ashviya: MicroSD Card - Store all variable data to MicroSD Card on microcontroller, allow for each run to be saved under a different filename (potentially in CSV format) so we can keep track of everything.
+
+#include <SD.h>
+#include <SPI.h>
+
+// Define pin numbers
+const int chipSelect = 4;
+
+// Define files
+File dataFile;
+String fileName;
+bool isFileNew = false; // Checks for new file
+
+// Define variables
+float temperatureC; // Raw temperature in Celcius
+float x_k; // Filtered temperature
+float zAngle; // z-axis angle
+float zAngleFiltered; // Filtered z-axis angle
+int runCount = 1; 
```

Comment:
There's a minor issue with the way you're incrementing the count. Everytime the mucrocontroller resets it'll start up at the top of your code, so it will reset it to 1 on every run overwriting the file. What you want to do is check how many files are already written to the sd card on startup (you can put this in the setup section) and then use that as your run count variable. Then the filename can be set to Run_ + that number and it will keep updating as the microcontroller is reset and runs again & again.